.github/workflows/release.yml

```diff
@@ -6,7 +6,10 @@ on:
       - published
 
 permissions:
+  attestations: write
   contents: read
+  id-token: write
+  packages: write
```

Comment:
I will move the permissions to specific job, but are descriptions really needed? Permissions are [documented and explained](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token), I don't see a value in providing the same info every time permission is used within multiple jobs. In most cases it can be inferred from the job context.

Reply:
I added descriptions anyway, but I am not sure if I am right about `id-token:write` 😅.

Reply:
> but I am not sure if I am right about `id-token:write` 😅.

Which is a good argument in favor of the descriptions. That makes it easy to check whether a given permission is still required.

`id-token:write` is also required for attestation. It's the permission that allows to turn the `GITHUB_TOKEN` into a JWT, which allows to prove to *external* consumers that the call is made from within GitHub Actions.

One example use case would be an AWS policy that allows these JWTs to access specific resources (e.g. to create an EC2 instance for deployment), without needing to manually store secrets.